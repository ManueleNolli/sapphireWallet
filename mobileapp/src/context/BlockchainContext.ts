import { createContext } from 'react'
import { NETWORKS } from '../constants/Networks'

type BlockchainContextType = {
  currentNetwork: NETWORKS
  setCurrentNetwork: (network: NETWORKS) => void
  ethersProvider: any
  setEthersProvider: (network: NETWORKS) => Promise<void>
}

export const BlockchainContext = createContext<BlockchainContextType>(<
  BlockchainContextType
>{
  currentNetwork: NETWORKS.LOCALHOST,
  setCurrentNetwork: (network: NETWORKS): void => {},
  ethersProvider: null,
  setEthersProvider: async (network: NETWORKS): Promise<void> => {},
})
