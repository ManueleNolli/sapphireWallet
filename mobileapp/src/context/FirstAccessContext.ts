import { createContext } from 'react'

type FirstAccessContextType = {
  isFirstAccess: boolean
  toggleFirstAccess: () => Promise<void>
}

export const FirstAccessContext = createContext<FirstAccessContextType>({
  isFirstAccess: true,
  toggleFirstAccess: async (): Promise<void> => {},
})
