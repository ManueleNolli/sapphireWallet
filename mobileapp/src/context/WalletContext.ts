import { createContext } from 'react'

type WalletContextType = {
  setPrivateKey: (privateKey: string) => Promise<void>
  getPrivateKey: (purposeMessage: string) => Promise<string>
  getEOAAddress: () => string
  setEOAAddress: (eoaAddress: string) => Promise<void>
  getWalletContractAddress: () => string
  setWalletContractAddress: (walletContractAddress: string) => Promise<void>
  resetWallet: () => Promise<void>
}

export const WalletContext = createContext<WalletContextType>(<
  WalletContextType
>{
  setPrivateKey: async (privateKey: string): Promise<void> => {},
  getPrivateKey: async (purposeMessage: string): Promise<string> => {
    console.warn('WalletContext.getPrivateKey not implemented')
    return Promise.resolve('')
  },
  getEOAAddress: (): string => {
    console.warn('WalletContext.getEOAAddress not implemented')
    return ''
  },
  setEOAAddress: async (eoaAddress: string): Promise<void> => {},
  getWalletContractAddress: (): string => {
    console.warn('WalletContext.getWalletContractAddress not implemented')
    return ''
  },
  setWalletContractAddress: async (
    walletContractAddress: string
  ): Promise<void> => {},
  resetWallet: async (): Promise<void> => {},
})
