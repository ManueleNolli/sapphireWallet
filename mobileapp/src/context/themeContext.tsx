import { Context, createContext } from 'react'

type ThemeContextType = {
  theme: string
  toggleTheme: () => void
}

export const ThemeContext: Context<ThemeContextType> = createContext({
  theme: 'light',
  toggleTheme: () => {},
})
