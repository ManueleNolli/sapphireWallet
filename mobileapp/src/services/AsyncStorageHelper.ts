import AsyncStorage from '@react-native-async-storage/async-storage'
export const storeData = async (key: string, value: string): Promise<void> => {
  await AsyncStorage.setItem(key, value).catch((e) => {
    console.error('Error storing data', e)
  })
}

export const getData = async (key: string): Promise<string | null> => {
  const value = await AsyncStorage.getItem(key).catch((e) => {
    console.error('Error getting data', e)
  })
  if (value !== null && value !== undefined) {
    return value
  }
  return null
}
