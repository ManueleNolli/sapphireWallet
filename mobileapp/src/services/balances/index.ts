export * from './balanceCrypto'
export * from './balanceNFTs'
