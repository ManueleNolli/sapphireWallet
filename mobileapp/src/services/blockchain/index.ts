export * from './utils'
export * from './OwnedNFTs'
