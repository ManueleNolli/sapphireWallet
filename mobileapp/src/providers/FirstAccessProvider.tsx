import React, { useEffect, useState } from 'react'
import useLoading from '../hooks/useLoading'
import { getData, storeData } from '../services/storage/'
import constants from '../constants/Constants'
import Loading from '../pages/Loading/Loading'
import { FirstAccessContext } from '../context/FirstAccessContext'
import FirstAccessStack from '../navigation/FirstAccessStack'

type FirstAccessProviderProps = {
  children: React.ReactNode
}

export function FirstAccessProvider({ children }: FirstAccessProviderProps) {
  const { isLoading, setIsLoading } = useLoading(true)
  const [isFirstAccess, setIsFirstAccess] = useState<boolean>(false)

  // load in local state for performance
  useEffect(() => {
    const initialiseFirstAccess = async () => {
      const firstAccess = await getData(constants.asyncStoreKeys.firstAccess)
      if (firstAccess === null || firstAccess === 'true') {
        setIsFirstAccess(true)
      }
    }

    initialiseFirstAccess().then(() => setIsLoading(false))
  }, [])

  const toggleFirstAccess = async () => {
    const nextValue = !isFirstAccess
    await storeData(constants.asyncStoreKeys.firstAccess, String(nextValue))
    setIsFirstAccess(!isFirstAccess)
  }

  if (isLoading) {
    return <Loading text={'Checking first access...'} />
  }

  return (
    <FirstAccessContext.Provider value={{ isFirstAccess, toggleFirstAccess }}>
      {isFirstAccess ? <FirstAccessStack /> : children}
    </FirstAccessContext.Provider>
  )
}
