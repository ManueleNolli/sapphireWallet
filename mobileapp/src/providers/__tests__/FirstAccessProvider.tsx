import React from 'react'
import { Text, Button } from 'react-native'
import { act, fireEvent, waitFor } from '@testing-library/react-native'
import { FirstAccessProvider } from '../FirstAccessProvider'
import { getData } from '../../services/storage/'
import { FirstAccessContext } from '../../context/FirstAccessContext'
import renderWithTheme from '../../TestHelper'

// MOCKS
const MockComponent = () => {
  return <Text>Mock</Text>
}

jest.mock('../../pages/Loading/Loading')
jest.mock('../../navigation/FirstAccessStack')
jest.mock('../../services/storage/')

describe('FirstAccessProvider', () => {
  it('Load First Access Stack', async () => {
    ;(getData as jest.Mock).mockReturnValue(Promise.resolve('true')) // firstAccess === true

    let tree

    await waitFor(async () => {
      tree = renderWithTheme(
        <FirstAccessProvider>
          <MockComponent />
        </FirstAccessProvider>
      )
    })

    expect(tree).toMatchSnapshot()
  })

  it('Load Children', async () => {
    ;(getData as jest.Mock).mockReturnValue(Promise.resolve(false))

    let tree

    await waitFor(async () => {
      tree = renderWithTheme(
        <FirstAccessProvider>
          <MockComponent />
        </FirstAccessProvider>
      )
    })

    expect(tree).toMatchSnapshot()
  })

  it('Toggle first access value', async () => {
    ;(getData as jest.Mock).mockReturnValue(Promise.resolve(false))

    const MockComponentToggle = () => {
      const { toggleFirstAccess } = React.useContext(FirstAccessContext)
      return <Button onPress={() => toggleFirstAccess()} title={'Toggle'} />
    }

    let tree

    await waitFor(async () => {
      tree = renderWithTheme(
        <FirstAccessProvider>
          <MockComponentToggle />
        </FirstAccessProvider>
      )
    })
    expect(tree).toMatchSnapshot()

    await act(async () => {
      const button = tree!.getByRole('button')
      fireEvent.press(button)
    })

    expect(tree).toMatchSnapshot()
  })

  it('Loading', async () => {
    ;(getData as jest.Mock).mockReturnValue(Promise.resolve(false))
    let tree = renderWithTheme(
      <FirstAccessProvider>
        <MockComponent />
      </FirstAccessProvider>
    )

    expect(tree).toMatchSnapshot()

    // Wait for loading
    await waitFor(() => {
      Promise.resolve()
    })
  })
})
