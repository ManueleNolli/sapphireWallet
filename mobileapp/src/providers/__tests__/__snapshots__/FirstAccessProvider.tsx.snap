// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstAccessProvider Load Children 1`] = `
<Text>
  Mock
</Text>
`;

exports[`FirstAccessProvider Load First Access Stack 1`] = `
<div
  data-testid="FirstAccessStack"
>
  FirstAccessStackMock
</div>
`;

exports[`FirstAccessProvider Loading 1`] = `
<div
  data-testid="Loading"
>
  Loading
</div>
`;

exports[`FirstAccessProvider Toggle first access value 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {},
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Toggle
    </Text>
  </View>
</View>
`;

exports[`FirstAccessProvider Toggle first access value 2`] = `
<div
  data-testid="FirstAccessStack"
>
  FirstAccessStackMock
</div>
`;
